test/test_encryption.py

```diff
@@ -1942,6 +1944,47 @@ def test_via_loading_shared_library(self):
         with self.assertRaises(ServerSelectionTimeoutError):
             no_mongocryptd_client.db.command("ping")
 
+    # https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/tests/README.rst#20-bypass-creating-mongocryptd-client-when-shared-library-is-loaded
+    @unittest.skipUnless(os.environ.get("TEST_CRYPT_SHARED"), "crypt_shared lib is not installed")
+    def test_client_via_loading_shared_library(self):
+        connection_established = [False]
```

Comment:
I think it would be more idiomatic to use a boolean, a lock, and a nonlocal, even if you are ensuring the thread is closed before accessing the value.

Reply:
Done.